lib/widgets/login/browser_login.dart

```diff
@@ -0,0 +1,181 @@
+import 'dart:math';
+import 'dart:typed_data';
+
+import 'package:convert/convert.dart';
+import 'package:drift/drift.dart';
+import 'package:flutter/widgets.dart';
+import 'package:url_launcher/url_launcher.dart';
+
+import '../../api/route/realm.dart';
+import '../../log.dart';
+import '../../model/store.dart';
+import '../app.dart';
+import '../login.dart';
+import '../store.dart';
+
+/// An InheritedWidget to co-ordinate the browser auth flow
+///
+/// The provided [navigatorKey] by this object should be attached to
+/// the main app widget so that when the browser redirects to the app
+/// using the universal link this widget can use it to access the current
+/// navigator instance.
+///
+/// This object also stores the temporarily generated OTP required for
+/// the completion of the flow.
+class BrowserLoginWidget extends InheritedWidget {
```

Comment:
This will need tests. Most critical is to have tests that cover pieces like the parsing logic in `loginFromExternalRoute`, which could be covered by small-scale unit tests.

But ideally we'd test more of the overall logic of how it fits together: so e.g. there'd be a test that starts out navigated to AuthMethodsPage, taps a button, and confirms we wind up on the appropriate HomePage. Generally I've been happy at how successful we've been in the Flutter app at writing widget tests at that level of complexity. So I'd like to try to have some kind of test at that level for this functionality too.

A key aspect of such a test is that it's not a full-blown integration test — those would *also* be nice, but are a further level of test complexity which we haven't yet done for anything and I don't expect as part of this issue. That means this test won't involve actual Android or iOS platform code; it'll all be within Dart. For some examples, see `test/notifications_test.dart`, and other tests that make heavy use of the bindings in `test/model/binding.dart`.